test/support/namespace.go

```diff
@@ -53,3 +53,34 @@ func deleteTestNamespace(t Test, namespace *corev1.Namespace) {
 	})
 	t.Expect(err).NotTo(gomega.HaveOccurred())
 }
+
+func CreateTestNamespaceWithName(t Test, namespaceName string, options ...Option[*corev1.Namespace]) *corev1.Namespace {
+	t.T().Helper()
+	namespace := &corev1.Namespace{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: corev1.SchemeGroupVersion.String(),
+			Kind:       "Namespace",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: namespaceName,
+		},
+	}
+
+	for _, option := range options {
+		t.Expect(option.applyTo(namespace)).To(gomega.Succeed())
+	}
+
+	namespace, err := t.Client().Core().CoreV1().Namespaces().Create(t.Ctx(), namespace, metav1.CreateOptions{})
+	t.Expect(err).NotTo(gomega.HaveOccurred())
+
+	return namespace
+}
+
+func DeleteTestNamespace(t Test, namespaceName string) {
+	t.T().Helper()
+	propagationPolicy := metav1.DeletePropagationBackground
```

Comment:
Can you add here:
```
	storeAllPodLogs(t, namespace)
	storeEvents(t, namespace)
```
after you rebase on latest main?

Reply:
Added method `StoreNamespaceLogs`  to reuse in multiple places as we are not going to delete namespace when the test `TestMNISTRayClusterUp` Passed

Reply:
a nitpick: what about using Namespace struct instead of namespaceName string? That way you don't need `GetNamespaceWithName` in `StoreNamespaceLogs`.

Reply:
Yes its a good idea, I will modify accordingly

Reply:
Added Namespace struct to initialise Namespace name and namespace instance